sqlglot/dialects/bigquery.py

```diff
@@ -144,6 +144,7 @@ class Parser(parser.Parser):
 
         FUNCTION_PARSERS = {
             **parser.Parser.FUNCTION_PARSERS,
+            "ARRAY": lambda self: self.expression(exp.Array, expressions=[self._parse_statement()]),
```

Comment:
why did you only add this for bigquery? i think this is generic

for example 

SELECT COALESCE((select y), 0)
FROM (
  select 1 as y
) z

works in postgres

Reply:
Discussed on slack.